lib/loader/ast/segment.cpp

```diff
@@ -99,16 +99,23 @@ Expect<void> Loader::loadSegment(AST::ElementSegment &ElemSeg) {
   // vec(expr): reference init list, vec(expr)
 
   // Read the checking byte.
-  EXPECTED_TRY(uint32_t Check, FMgr.readU32().map_error([this](auto E) {
-    return logLoadError(E, FMgr.getLastOffset(), ASTNodeAttr::Seg_Element)
-        .value();
-  }));
-  // Check > 0 cases are for BulkMemoryOperations or ReferenceTypes proposal.
-  if (Check > 0 && !Conf.hasProposal(Proposal::BulkMemoryOperations) &&
-      !Conf.hasProposal(Proposal::ReferenceTypes)) {
-    return logNeedProposal(ErrCode::Value::ExpectedZeroByte,
-                           Proposal::BulkMemoryOperations, FMgr.getLastOffset(),
-                           ASTNodeAttr::Seg_Element);
+  uint32_t Check = 0;
```

Comment:
I think the comment is outdated then?

Reply:
Here is still for the checking byte, but just handle the special case.

Reply:
No, I mean `// Read the checking byte.` no longer exists?

Reply:
It exists, you can find it in the `else` statement.

Reply:
I see